### ISSUE TITLE
Port patterns from 26_denial_of_service.rule

### ISSUE THREAD

**Issue by dmajda** [openSUSE/scanny#31]
We need to port patterns form the `rules/26_denial_of_service.rule` file from the [old scanner](http://gitorious.org/code-scanner/ror-sec-scanner/):

```
Desc: Denial of Service

# impact        CWE identifier          regex
medium          CWE-400                 \.find[\w_]*\s*[\(]*.*(conditions|limit)\s*=>.*LIKE
```

**Comment by dmajda** [comment]
@LTe I'm looking this part of the pattern code in `DenialOfServiceCheck`:

```
HashLiteral<
  array = [
    any*,
    SymbolLiteral<value = :limit | :conditions>,
    StringLiteral<string *= /LIKE/>,
    any*
  ]
>,
```

As it is written, it will not only match `:limit => "...LIKE..."` or `:conditions => "...LIKE..."`, but also `"...LIKE..." => :limit` and `"...LIKE..." => :conditions`. This is because the pattern doesn't check for position of the key and value patterns in `HashLiteral`'s array.

This problem can be solved by rewriting `any*` in the array pattern to `any{even}`. See for example `SetSecretCheck` where this is done.

Could you fix this pattern and also all other instances of this problem in other patterns?

**Comment by dmajda** [comment]
The hash issues were fixed in #74. The pattern is ported in `DenialOfServiceCheck`, closing.
